components/maze_solver/include/maze_solver/direction.h

```diff
@@ -213,6 +213,27 @@ constexpr auto to_radians(Direction d) noexcept
     }
 }
 
+template <ExtendedFloatingPoint F = float>
+constexpr auto to_direction(const F &a) noexcept
```

Comment:
The default type is redundant, the only value it gives is making passing `to_degrees<>` as a function pointer valid (instead of forcing `to_degrees<float>`:

```suggestion
template <ExtendedFloatingPoint F>
constexpr auto to_direction(const F &a) noexcept
```

But in this case, I think an implicit template can also work:

```suggestion
constexpr auto to_direction(const ExtendedFloatingPoint auto &a) noexcept
```

The documentation is also missing.